### PR TITLE
Fix limited block fail message priority

### PR DESCRIPTION
**Pull request by joshlory**
PR https://github.com/code-dot-org/code-dot-org/pull/19678 set the wrong precedence for which failure message shows up when the goal is not met.

Broken:

image

Fixed:

image

### CODE REVIEW

File: apps/src/maze/bee.js [hunk 1]
```diff
@@ -192,8 +192,8 @@ export default class Bee extends Gatherer {
         // If we have a non-app specific failure, we want that to take precedence.
         // Values over TOO_MANY_BLOCKS_FAIL are not true failures, but indicate
         // a suboptimal solution, so in those cases we want to return our
-        // app specific fail
-        if (testResults >= TestResults.TOO_MANY_BLOCKS_FAIL) {
+        // app specific fail. Same goes for BLOCK_LIMIT_FAIL.
+        if (testResults >= TestResults.TOO_MANY_BLOCKS_FAIL || testResults === TestResults.BLOCK_LIMIT_FAIL) {
```
Comment (Hamms) [inline apps/src/maze/bee.js:196]: Why should `BLOCK_LIMIT_FAIL` be an app-specific fail?

Comment (joshlory) [reply apps/src/maze/bee.js:196]: In this case we fail for multiple reasons: too many "maze_moveForward" blocks were used, but they also didn't harvest any pumpkins.  I don't fully understand how the priority of app-specific vs. generic failure messages work 😕.

My thought process was that I wanted the `DID_NOT_COLLECT_EVERYTHING` error message to take precedence over the `BLOCK_LIMIT_FAIL` feedback message.  Is there a better way to do this?

Comment (Hamms) [reply apps/src/maze/bee.js:196]: Ahhh, gotcha. Yeah, the relative priorities of failures are complicated and constantly getting more so.

I think this is a new category; legitimate failures that should have a lower priority than app-specific failures. I'm not sure what the better way to do this is, but I'd definitely like to see it done in a way that'll accommodate additions to this category.

Also, why was this added just for these three subtypes and not for all of them?

Comment (joshlory) [reply apps/src/maze/bee.js:196]: Followed up offline: this unwinds the existing special-case logic for these 3 apps.  Not super clean, but should be ok to merge.

We should clean up how feedback messages are prioritized between generic and app-specific failures (added tech debt item).

**Pull request merged**